ext/pcntl/php_signal.c

```diff
@@ -43,7 +43,7 @@ Sigfunc *php_signal4(int signo, Sigfunc *func, int restart, int mask_all)
 #ifdef HAVE_STRUCT_SIGINFO_T
 	act.sa_flags |= SA_SIGINFO;
 #endif
-	if (signo == SIGALRM || (! restart)) {
+	if (signo == SIGALRM && (! restart)) {
```

Comment:
Rather than changing this to an `&&`, shouldn't the `signo == SIGALRM` part just be dropped? We still want to allow using SA_RESTART with non-SIGALRM signals.

Reply:
Yes please see [my original intention](https://github.com/Microsoft/msphpsql/issues/885#issuecomment-451001996) but it might be drastic and break other existing code.